app/waf/test/acceptance/application.controller.acceptance.ts

```diff
@@ -218,13 +218,68 @@ describe('ApplicationController', () => {
       .expect(204);
   });
 
+  it(
+    'post ' +
+      prefix +
+      '/applications/{id}/deploy: deploy an unspported action.',
+    async () => {
+      let adc = await givenAdcData(wafapp);
+      let application = await givenApplicationData(wafapp, {
+        adcId: adc.id,
+      });
+
+      let response = await client
+        .post(prefix + '/applications/' + application.id + '/declarations')
+        .set('X-Auth-Token', ExpectedData.userToken)
+        .set('tenant-id', ExpectedData.tenantId)
+        .send({name: 'a-declaration'})
+        .expect(200);
+
+      await client
+        .patch(prefix + '/applications/' + application.id)
+        .set('X-Auth-Token', ExpectedData.userToken)
+        .set('tenant-id', ExpectedData.tenantId)
+        .send({defaultDeclarationId: response.body.declaration.id})
+        .expect(204);
+
+      deployStub.throws(
+        Object('InvalidPatchOperationError: path does not exist'),
+      );
+
+      await client
+        .post(prefix + '/applications/' + application.id + '/deploy')
+        .set('X-Auth-Token', ExpectedData.userToken)
+        .set('tenant-id', ExpectedData.tenantId)
+        .expect(500);
+
+      deployStub.throws(Object('test error'));
+
+      await client
+        .post(prefix + '/applications/' + application.id + '/deploy')
+        .set('X-Auth-Token', ExpectedData.userToken)
+        .set('tenant-id', ExpectedData.tenantId)
+        .expect(200);
+
+      deployStub.returns(Promise.resolve('Hello'));
+
+      await client
+        .post(prefix + '/applications/' + application.id + '/deploy')
+        .set('X-Auth-Token', ExpectedData.userToken)
+        .set('tenant-id', ExpectedData.tenantId)
+        .expect(200);
+    },
+  );
+
   it(
     'post ' + prefix + '/applications/{id}/deploy: deploy application',
     async () => {
       let adc = await givenAdcData(wafapp);
       let application = await givenApplicationData(wafapp, {
         adcId: adc.id,
       });
+      let applicationNew = await givenApplicationData(wafapp, {
```

Comment:
I am not able to understand the purpose of this test...

Reply:
The purpose is test the two types of exceptions, 1) the exception with error string "InvalidPatchOperationError: path does not existInvalidPatchOperationError: path does not exist", 2) the exception with error string that isn't "InvalidPatchOperationError: path does not exist".